fuzz/fuzz_targets/serial.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_main]
+use devices::legacy::Serial;
+use libc::EFD_NONBLOCK;
+use libfuzzer_sys::fuzz_target;
+use std::sync::Arc;
+use vm_device::interrupt::{InterruptIndex, InterruptSourceConfig, InterruptSourceGroup};
+use vm_device::BusDevice;
+use vmm_sys_util::eventfd::EventFd;
+
+fuzz_target!(|bytes| {
+    // Need multiples of 4 bytes to fill out the test
+    if bytes.len() % 4 != 0 {
+        return;
+    }
+
+    let mut serial = Serial::new_sink(
+        "serial".into(),
+        Arc::new(TestInterrupt::new(EventFd::new(EFD_NONBLOCK).unwrap())),
+    );
+
+    let mut i = 0;
+    while i < bytes.len() {
+        let offset = i;
+        let data = i + 1;
+        serial.write(0, bytes[offset].into(), &bytes[data..data + 1]);
```

Comment:
It may worth constraining the the input `offset` to `write/read` to be valid, e.g. `bytes[offset].into() % 8`, because we know the invalid `offset` will simply return `0` or `Ok`.